File: stalinSort/stalingSort.cpp
#include <vector>

std::vector<int> stalinSort(std::vector<int> arr)
{
    if (arr.size() == 0)
    {
        return arr;
    }

    int firstElement = arr[0];
    std::vector<int> sorted_list = { firstElement };

    for (int i = 1; i < arr.size(); i++)
    {
        if (arr[i] >= sorted_list[sorted_list.size() - 1])
        {
            sorted_list.push_back(arr[i]);
        }
    }
    return sorted_list;
}

File: stalinSort/stalingSort.js
function stalinSort(arr)
{
    if(arr.length === 0) {return [];}

    let sorted_list = [arr[0]];
    
    for(let i = 1; i < arr.length; i++)
    {
        if(arr[i] >= sorted_list[sorted_list.length - 1])
        {
            sorted_list.push(arr[i]);
        }
    }
    return sorted_list;
}